### PR TITLE
Implements Azure deploy for Linux instances.

### PR DESCRIPTION
**Pull request by AliceAlbano**

### CODE REVIEW

**Comment by emuus** [comment]
ok for me !

File: hammr/utils/deployment_utils.py [hunk 1]
```diff
@@ -233,4 +274,32 @@ def show_deploy_progress_openstack(image_object, deployed_instance_id, bar_statu
     bar_status.message = "Deployment complete"
     progress.update(bar_status.percentage)
     progress.finish()
-    return status
\ No newline at end of file
+    return status
+
+def query_password_azure(question):
```
Comment (ludop-usharesoft) [inline hammr/utils/deployment_utils.py]: Could be a generic method to ask password (remove "azure" from method name)

File: hammr/utils/deployment_utils.py [hunk 2]
```diff
@@ -90,6 +92,44 @@ def build_deployment_amazon(file):
...
+        printer.out("There is no attribute [name] for the provisioner", printer.ERROR)
+        return None
+    deployment.name = file["name"]
+    if "userName" in file:
+        myinstance.userName = file["userName"]
+    else:
+        printer.out("There is no attribute [userName] for the provisioner", printer.ERROR)
+        return None
+    if "userSshKey" in file:
+        myinstance.userSshKey = file["userSshKey"]
```
Comment (ludop-usharesoft) [inline hammr/utils/deployment_utils.py]: No regexp validation on the sshKey enter by the user but I think it's ok.

File: hammr/utils/deployment_utils.py (hunk 1 above)
Comment (AliceAlbano) [reply hammr/utils/deployment_utils.py]: At first I made a generic method, but inside the method I used a regexp to check the password, and it is specific to azure. So I think it should stay specific to azure, what do you think ?

File: hammr/utils/deployment_utils.py [hunk 3]
```diff
@@ -233,4 +274,32 @@ def show_deploy_progress_openstack(image_object, deployed_instance_id, bar_statu
...
+        if valid_input == 0:
+            first_choice = getpass.getpass(prompt=question)
+            if re.match(pattern, first_choice):
+                valid_input += 1
+            else:
+                printer.out("""The user password must be between 6-72 characters long and must contains at least one uppercase character, one lowercase character, one numeric digit and one special character (@#$%^&+=*!)""", printer.WARNING)
+        elif valid_input == 1:
+            second_choice = getpass.getpass(prompt="Please confirm your password: ")
+            if second_choice == first_choice:
+                valid_input += 1
```
[the remark this answers is not in the document]
Comment (AliceAlbano) [reply hammr/utils/deployment_utils.py]: Thank you, I wonder where I got my ideas sometimes ;)

File: hammr/utils/deployment_utils.py (hunk 1 above)
Comment (ludop-usharesoft) [reply hammr/utils/deployment_utils.py]: ok

**Pull request closed without merging**